packages/ui/src/ui/store/selectors/groups.ts

```diff
@@ -88,15 +88,33 @@ const getGroupsTreeFiltered = createSelector(
 );
 
 const getGroupsTreeFilteredAndExpanded = createSelector(
-    [getGroupsTreeFiltered, getGroupsExpanded],
-    (root, expanded) => {
-        const res = cloneDeep_(root);
+    [getGroupsTree, getGroupsTreeFiltered, getGroupsExpanded, getGroupsNameFilter],
+    (groupTree, groupsTreeFiltered, expandedByUser, groupNameFilter) => {
+        const expanded: Record<string, boolean> = {
+            ...expandedByUser,
+        };
+
+        const res = cloneDeep_(groupsTreeFiltered);
+
+        hammer.treeList.treeForEach(res.children, (node: GroupsTreeNode) => {
+            const isNodeMatchedFilter = groupNameFilter && node.name.includes(groupNameFilter);
+
+            if (isNodeMatchedFilter) {
+                let parentName = node.parent!;
+
+                while (groupTree[parentName]) {
+                    expanded[parentName] = true;
```

Comment:
The changes enforce expanding of parents. What should user do if he wants to collapse an unnecessary branch?

Reply:
Fixed, now user can do that